### ISSUE TITLE
CMake build fails

### ISSUE THREAD

**Issue by paoloviviani** [alpha-unito/gam#37]
CMake build fails on Ubuntu 17.10 with the following error

```
[ 15%] Linking CXX executable simple_public
/usr/bin/ld: CMakeFiles/simple_public.dir/simple_public.cpp.o: undefined reference to symbol 'pthread_create@@GLIBC_2.2.5'
//lib/x86_64-linux-gnu/libpthread.so.0: error adding symbols: DSO missing from command line
```

Build by:
```
$ cd gam
$ mkdir build
$ cd build
$ cmake -DLIBFABRIC_LIBRARY_DIRS=/usr/lib/x86_64-linux-gnu/ -D LIBFABRIC_INCLUDE_DIRS=/usr/include/rdma/ ..
$ make
```

**Comment by droccom** [comment]
I probably forgot to link the pthread library. I will fix it.

Also, I think that the current cmake would just overwrite the variables you are passing. If you really need to specify two "independent" paths for includes and libraries, I will modify the cmake. Otherwise, you should be able to set a libfabric path as: `LIBFABRIC_ROOT` (that is the standard cmake way). Let me know.

**Comment by paoloviviani** [comment]
I'll investigate. But in case of Ubuntu and apt-installed libfabric,
defining a root is tricky. Still, it should be the only case where I need
to specify both.



Il giorno mar 8 gen 2019, 18:32 Maurizio Drocco <notifications@github.com>
ha scritto:

**Comment by droccom** [comment]
The current cmake module for libfabric is actually a wrapper of pkg-config, so I think the root is used just to locate the file `lib/pkgconfig/libfabric.pc`.

**Comment by paoloviviani** [comment]
I see!
I'll try that solution.


Il giorno mar 8 gen 2019, 18:40 Maurizio Drocco <notifications@github.com>
ha scritto:

**Comment by droccom** [comment]
> CMake build fails on Ubuntu 17.10 with the following error
> 
> ```
> [ 15%] Linking CXX executable simple_public
> /usr/bin/ld: CMakeFiles/simple_public.dir/simple_public.cpp.o: undefined reference to symbol 'pthread_create@@GLIBC_2.2.5'
> //lib/x86_64-linux-gnu/libpthread.so.0: error adding symbols: DSO missing from command line
> ```

That should be fixed by #41, please check.

**Comment by paoloviviani** [comment]
Build and tests works nicely now.

Still, libfabric paths needs to be specified explicitly, as the apt-installed version puts the libfabric.pc in a path that is not reachable by specifying LIBFABRIC_ROOT. Namely, the path for Ubuntu 17.10 is
```
/usr/lib/x86_64-linux-gnu/pkgconfig/libfabric.pc
```
I don't see this as extremely critical, so I would close the issue. However, I would provide a small README with cmake options available (like libfabric paths). I can do it later.
